src/Service/Reports/SernosUsedOnInvoiceModule.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace Linn.Products.Service.Reports
+{
+    using Linn.Products.Facade.Services;
+    using Linn.Products.Resources;
+    using Linn.Products.Service.Models;
+
+    using Nancy;
+    using Nancy.ModelBinding;
+
+    public sealed class SernosUsedOnInvoiceModule : NancyModule
+    {
+        private readonly ISernosUsedOnInvoiceFacade facadeService;
+
+        public SernosUsedOnInvoiceModule(ISernosUsedOnInvoiceFacade facadeService)
+        {
+            this.facadeService = facadeService;
+            this.Get("/products/reports/sernos-used-on-invoice", parameters => this.GetSernosUsedOnInvoice());
+        }
+
+        private object GetSernosUsedOnInvoice()
+        {
+            var resource = this.Bind<SernosUsedOnInvoiceReportRequestResource>();
+         
```

Comment:
You can validate the resource you've got here using a Validator (FluentValidation) there are some examples in Resources/Validators and the corresponding modules for those classes e.g TariffModule

Reply:
👍